[Aditi] iP

aditi2313 opened the pull request:
### DukePro

> _“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))_

DukePro frees your mind of having to remember things you need to do. It's:
- text-based
- easy to learn
- _SUPER FAST_ to use

All you need to do is,

1. Download it from [here](https://github.com/aditi2313/ip).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉
And it is **FREE**! 😱🥳

Features:
- [x]  Managing tasks
- [x]  Managing deadlines
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public static void main(String[] args) {
  System.out.println("Hello! I'm Duke :) \nWhat can I do for you? :D");
   new Duke("data.txt").run();
}
```

Commit:
Add Gradle support

aditi2313 committed:
Levela 1,2,3,4,5,6

aditi2313 committed:
Level 1,2,3,4,5,6

aditi2313 committed:
Level 1,2,3

aditi2313 committed:
Task class

aditi2313 committed:
Level 1, 2, 3, 4

aditi2313 committed:
Task class

aditi2313 committed:
Todo class

aditi2313 committed:
Deadline class

aditi2313 committed:
Event class

aditi2313 committed:
Levels 1, 2, 3, 4, 5, 6

aditi2313 committed:
Duke Exception

aditi2313 committed:
input tests

aditi2313 committed:
expected outputs

aditi2313 committed:
Duke main file

aditi2313 committed:
Duke main file

aditi2313 committed:
Duke main file

aditi2313 committed:
Update Duke.java

aditi2313 committed:
Create data.txt

aditi2313 committed:
Update Duke.java

aditi2313 committed:
Update Deadline.java

aditi2313 committed:
Update Event.java

aditi2313 committed:
Update Todo.java

aditi2313 committed:
Update Task.java

aditi2313 committed:
Create data.txt

aditi2313 committed:
Merge pull request #1 from aditi2313/branch-Level-7

Branch level 7

aditi2313 committed:
Update Duke.java

aditi2313 committed:
Update Duke.java

aditi2313 committed:
Merge branch 'master' into branch-Level-8

aditi2313 committed:
Merge pull request #2 from aditi2313/branch-Level-8

Branch level 8

aditi2313 committed:
Duke main file

aditi2313 committed:
Duke main file

aditi2313 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/Duke.java

aditi2313 committed:
Duke main file

aditi2313 committed:
A-JUnit Testing

aditi2313 committed:
A-Jar

aditi2313 committed:
Update Duke.java

aditi2313 committed:
Update TaskList.java

aditi2313 committed:
Update Todo.java

aditi2313 committed:
Update Storage.java

aditi2313 committed:
Update Parser.java

aditi2313 committed:
Update DukeException.java

aditi2313 committed:
Update Ui.java

aditi2313 committed:
Update Deadline.java

aditi2313 committed:
Update Event.java

aditi2313 committed:
Update Task.java

aditi2313 committed:
Update DukeException.java

aditi2313 committed:
Update Parser.java

aditi2313 committed:
Update Storage.java

aditi2313 committed:
Update Deadline.java

aditi2313 committed:
Update Event.java

aditi2313 committed:
Update Task.java

aditi2313 committed:
Update TaskList.java

aditi2313 committed:
Update Todo.java

aditi2313 committed:
Update Ui.java

aditi2313 committed:
Update Duke.java

aditi2313 committed:
Update TaskList.java

aditi2313 committed:
Update Parser.java

aditi2313 committed:
Merge pull request #3 from aditi2313/branch-Level-9

Branch level 9

aditi2313 committed:
Merge pull request #4 from aditi2313/branch-A-CodingStandard

Branch a coding standard

aditi2313 committed:
Merge branch 'master' into branch-A-JavaDoc

aditi2313 committed:
Merge pull request #5 from aditi2313/branch-A-JavaDoc

Branch a java doc

aditi2313 committed:
Latest commit Level-9

ongkimlai commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,54 @@
+     * @throws DukeException If user input message does not make sense.
+     */
+    public void userCommand(String[] inputStringsArray, Storage storage) throws DukeException {
+        switch (inputStringsArray[0]) {
+            case "list":
+                tasks.displayList();
+                break;
+            case "mark":
+                tasks.mark(inputStringsArray[1]);
+                storage.updateData(tasks);
+                break;
```
Remember to have no indentations for the case clauses. 😃 

```suggestion
        switch (inputStringsArray[0]) {
        case "list":
            tasks.displayList();
            break;
        case "mark":
            tasks.mark(inputStringsArray[1]);
            storage.updateData(tasks);
            break;
```

shurvirarora replied:
Good spot! You can manually change this IntelliJ to make it default.

aditi2313 replied:
Ah! Thanks for that. Will do!

ongkimlai commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,54 @@
+                tasks.unmark(inputStringsArray[1]);
+                storage.updateData(tasks);
+                break;
+            case "todo":
+            case "deadline":
+            case "event":
+                tasks.addToList(inputStringsArray);
```
Remember to leave a `// Fallthrough` whenever there is a case statement without a break statement.

aditi2313 replied:
Oh alright! Will make that change :)

ongkimlai reviewed (commented):
Looks good overall 👍 Just keep in my the coding standards for the switch-case statements.

shurvirarora commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,143 @@
+        int num = Integer.parseInt(number);
+        tasks.get(num - 1).setAsDone();
+        Task temp = tasks.get(num - 1);
+        System.out.println("Nice! :P I've marked this task as done:\n [" + temp.symbol() + "][" +
```
perhaps a line break before the operator "+" could make the code look neater 👍

aditi2313 replied:
Oh I see! Thanks for that suggestion :))

shurvirarora commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,143 @@
+        String returnString = "Here are the matching tasks in your list:\n";
+        for (int i = 0; i < foundTasks.size(); i++) {
+            returnString = returnString + (i + 1) + ". [" + foundTasks.get(i).symbol() + "]["
+                    + foundTasks.get(i).getStatusIcon() + "] " + foundTasks.get(i).displayTime() + "\n";
```
Line break for long lines would make the code look neater 👍  (e.g. before operator "+")

shurvirarora commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,48 @@
+     */
+    public void startConversation(Parser parser, Storage storage) throws DukeException {
+        Scanner sc = new Scanner(System.in);
+        String inputString = sc.nextLine();
```
Perhaps the naming of this could be "stringInput" to emphasise that it is referring to an input, that is of String type - or simply "input" since we have already established that it is of type String. 👍

aditi2313 replied:
Hm I see... Thanks!

shurvirarora commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,57 @@
+ */
+abstract public class Task {
+    protected String details;
+    protected boolean isDone;
```
Good naming of boolean variable with use of "is" 👍

Shivang211 replied:
Agreed! Nicely done 👍

shurvirarora reviewed (commented):
Overall code LGTM! Slight nits to fix such as line breaks for long lines.

aditi2313 committed:
Merge pull request #6 from aditi2313/add-gradle-support

Add Gradle support

aditi2313 committed:
Update Duke.java

aditi2313 committed:
Duke main

aditi2313 committed:
Update Duke.java

aditi2313 committed:
Merge branch 'master' of https://github.com/aditi2313/ip

aditi2313 committed:
Duke main

Shivang211 commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,143 @@
+        int num = Integer.parseInt(number);
+        tasks.get(num - 1).setAsNotDone();
+        Task temp = tasks.get(num - 1);
+        System.out.println("OK ._. , I've marked this task as not done yet:\n [" + temp.symbol() + "][" +
```
Good job on avoiding lines more than 100 chars!! Nicely broken down into two lines

Shivang211 commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,78 @@
+            while ((line = br.readLine()) != null) {
+                String[] lineArray = line.split("/");
+                Task task;
+                switch (lineArray[0]) {
```
Indentation for switch is incorrect here, can update the code editor settings as per this module's coding standards

aditi2313 replied:
Yup! Will do :) Thanks!

aditi2313 committed:
JavaFx

aditi2313 committed:
Delete hs_err_pid42578.log

aditi2313 committed:
Delete hs_err_pid42578.log

aditi2313 committed:
Merge pull request #7 from aditi2313/branch-Level-10

Level-10

aditi2313 committed:
JavaFx latest

aditi2313 committed:
Update data.txt

aditi2313 committed:
JavaFx latest

aditi2313 committed:
JavaFx latest

aditi2313 committed:
JavaFx latest

aditi2313 committed:
Updated Duke class without unused methods like (eg. greetUser())

Method functionality was added to initialise() in MainWindow class. Unused
methods may cause confusion in the future.

Let's remove the method.

Doing so will make debugging easier when more code is added to the
class and also cause lesser confusion and improve readability.

aditi2313 committed:
Updated JavaFx

aditi2313 committed:
Updated MainWindow class initialize() method.

greetUser() method functionality added from Duke class
to initialize() method. Adding the label message here
makes implementation easier.

Let's add the user greeting directly to the label in the
method.

Doing so will make debugging easier, cause lesser confusion
and improve readability.

aditi2313 committed:
Updated MainWindow class initialize() method.

greetUser() method functionality added from Duke class
to initialize() method. Adding the label message here
makes implementation easier.

Let's add the user greeting directly to the label in the
method.

Doing so will make debugging easier, cause lesser confusion
and improve readability.

aditi2313 committed:
Updated Duke class without unused methods (eg. greetUser(), getDialogLabel())

greetUser() method functionality was added to initialise() in MainWindow class.
getDialogLabel() method was unused. Unused methods may cause confusion in the future.

Let's remove the methods.

Doing so will make debugging easier when more code is added to the
class and also cause lesser confusion and improve readability.

aditi2313 committed:
Duke class with assertions

aditi2313 committed:
Update Storage.java

aditi2313 committed:
Update TaskList.java

aditi2313 committed:
Update Parser.java

aditi2313 committed:
Merge pull request #8 from aditi2313/branch-A-Assertions

Duke class with assertions

aditi2313 committed:
Merge pull request #9 from aditi2313/master

Merge from master branch

aditi2313 committed:
Merge pull request #10 from aditi2313/branch-A-CodeQuality

Branch code quality

aditi2313 committed:
Updated with code quality checks

aditi2313 committed:
Updated with code quality checks

aditi2313 committed:
Updated MainWindow class initialize() method.

greetUser() method functionality added from Duke class
to initialize() method. Adding the label message here
makes implementation easier.

Let's add the user greeting directly to the label in the
method.

Doing so will make debugging easier, cause lesser confusion
and improve readability.

aditi2313 committed:
Updated Duke class without unused method greetUser()

greetUser() method functionality added to MainWindow class
to initialize() method. Removing unused methods makes code
more systematic and concise.

Let's add remove the greetUser() method.

Doing so will make debugging easier, cause lesser confusion
and improve readability.

aditi2313 committed:
Updated Duke class without unused method greetUser()

greetUser() method functionality added to MainWindow class
to initialize() method. Removing unused methods makes code
more systematic and concise.

Let's add remove the greetUser() method.

Doing so will make debugging easier, cause lesser confusion
and improve readability.

aditi2313 committed:
Updated Duke class without unused method greetUser()

greetUser() method functionality added to MainWindow class
to initialize() method. Removing unused methods makes code
more systematic and concise.

Let's add remove the greetUser() method.

Doing so will make debugging easier, cause lesser confusion
and improve readability.

aditi2313 committed:
Updated Duke class with new method greetUser()

greetUser() method functionality added to Duke class
allows for abstraction when the bot is started and a
welcome message needs to be displayed. This makes the
MainWindow class SLAP hard.

Let's add a greetUser() method which returns a string
greeting.

Doing so will make increase readability of the code
and makes debugging easier in the future. Also ensures
that relevant information is abstracted away.

aditi2313 committed:
Updated MainWindow class initialize() method.

greetUser() method called added from Duke object.
Calling the method from within the Label makes
implementation easier and also allows for better abstraction.

Let's call the greetUser() method from the Duke class within
the label.

Doing so will make debugging easier, cause lesser confusion
and improve readability. Also means that the code SLAPs harder.

aditi2313 committed:
Updated MainWindow class

aditi2313 committed:
Updated MainWindow class initialize() method.
greetUser() static method called from Ui class.
Makes implementation easier and also allows for
better abstraction.

Let's call the greetUser() method from the Ui class.

Doing so will make debugging easier, cause lesser confusion
and improve readability. Also means that the code SLAPs harder.

aditi2313 committed:
Updated Ui class with user greeting functionality.

greetUser() method functionality added to Ui class.
Adding greeting here makes code more systematic and
concise.

Let's add a static greetUser() method.

Doing so will make debugging easier, cause lesser confusion
and improve readability.

aditi2313 committed:
Updated Duke class.

aditi2313 committed:
BCD Extension additions

aditi2313 committed:
Updated MainWindow class initialize() method with BCD Extension.
greetUser() static method called from Ui class.
Makes implementation easier and also allows for
better abstraction.

Let's call the greetUser() method from the Ui class.

Doing so will make debugging easier, cause lesser confusion
and improve readability. Also means that the code SLAPs harder.

aditi2313 committed:
Updated Ui class with user greeting functionality and BCD Extension.
greetUser() method functionality added to Ui class.
Adding greeting here makes code more systematic and
concise.

Let's add a static greetUser() method.

Doing so will make debugging easier, cause lesser confusion
and improve readability.

aditi2313 committed:
Updated Ui class with user greeting functionality and BCD Extension.
greetUser() method functionality added to Ui class.
Adding greeting here makes code more systematic and
concise.

Let's add a static greetUser() method.

Doing so will make debugging easier, cause lesser confusion
and improve readability.

aditi2313 committed:
Data file for note functionality

aditi2313 committed:
Data file for note functionality

aditi2313 committed:
Data file for note functionality

aditi2313 committed:
Improved GUI

aditi2313 committed:
Update MainWindow class initialize() method and add BCD Extension.
greetUser() static method called from Ui class.
Makes implementation easier and also allows for
better abstraction.

Let's call the greetUser() method from the Ui class.

Doing so will make debugging easier, cause lesser confusion
and improve readability. Also means that the code SLAPs harder.

aditi2313 committed:
Update Ui class with user greeting functionality and add BCD Extension.
greetUser() method functionality added to Ui class.
Adding greeting here makes code more systematic and
concise.

Let's add a static greetUser() method.

Doing so will make debugging easier, cause lesser confusion
and improve readability.

aditi2313 committed:
Update README.md

aditi2313 committed:
User Guide updates

aditi2313 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	docs/README.md

aditi2313 committed:
Update README.md

aditi2313 committed:
Update Ui class with user greeting functionality and add BCD Extension.
greetUser() method functionality added to Ui class.
Adding greeting here makes code more systematic and
concise.

Let's add a static greetUser() method.

Doing so will make debugging easier, cause lesser confusion
and improve readability.

aditi2313 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	docs/README.md

aditi2313 committed:
Update README.md

aditi2313 committed:
User Guide

aditi2313 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	docs/README.md

aditi2313 committed:
User Guide

aditi2313 committed:
User Guide

aditi2313 committed:
User Guide

aditi2313 committed:
A better GUI

aditi2313 committed:
Set theme jekyll-theme-cayman

aditi2313 committed:
Update README.md

aditi2313 committed:
Creating JAR file

aditi2313 committed:
Merge remote-tracking branch 'origin/master'

aditi2313 committed:
Creating JAR file

aditi2313 committed:
Bug fixes

aditi2313 committed:
Checkstyle

aditi2313 committed:
Update MainWindow class initialize() method and add BCD Extension.
greetUser() static method called from Ui class.
Makes implementation easier and also allows for
better abstraction.

Let's call the greetUser() method from the Ui class.

Doing so will make debugging easier, cause lesser confusion
and improve readability. Also means that the code SLAPs harder.

aditi2313 committed:
Update Ui class with user greeting functionality and add BCD Extension.
greetUser() method functionality added to Ui class.
Adding greeting here makes code more systematic and
concise.

Let's add a static greetUser() method.

Doing so will make debugging easier, cause lesser confusion
and improve readability.

aditi2313 committed:
Checkstyle

aditi2313 committed:
Checkstyle

aditi2313 committed:
Bug fixes

aditi2313 committed:
Bug fixes

aditi2313 committed:
Update Ui class with user greeting functionality and add BCD Extension.
greetUser() method functionality added to Ui class.
Adding greeting here makes code more systematic and
concise.

Let's add a static greetUser() method.

Doing so will make debugging easier, cause lesser confusion
and improve readability.

aditi2313 committed:
Improved javadoc comments

aditi2313 committed:
Update Ui class with user greeting functionality and add BCD Extension.
greetUser() method functionality added to Ui class.
Adding greeting here makes code more systematic and
concise.

Let's add a static greetUser() method.

Doing so will make debugging easier, cause lesser confusion
and improve readability.

aditi2313 committed:
bug fixes

aditi2313 committed:
Bug fixes

aditi2313 committed:
Final User Guide

aditi2313 committed:
Final User Guide

aditi2313 committed:
Task List
